oslo_aero_3_0a003: Feature/add leung dust emis

Ovewh opened the pull request:
Update of draft PR #43  That resolve merge conflicts  in the implementation to resolve #42 . It has not been tested yet or compiled. I'm happy to do that if the reviewers think the implementation is correct.

resolves #42 
supersedes #43

mvertens committed:
draft implementation of how to adden leung dust emissions

Commit:
Update resolved conflict with CAM updates since initial PR was created

Have not tested if it complies

oyvindseland reviewed (approved):
Minor detail in readnl  Leung_2003 should be corrected to Leung_2023

gold2718 commented on src/oslo_aero_dust.F90:
```diff
@@ -161,30 +159,37 @@ subroutine oslo_aero_dust_emis(lchnk, ncol, dstflx, cflx)
-         totalEmissionFlux(icol) = totalEmissionFlux(icol) + sum(dstflx(icol,:))
-      end do
-
-      ! Note that following CESM use of "dust_emis_fact", the emissions are
-      ! scaled by the INVERSE of the factor!!
-      ! There is another random scale factor of 1.15 there. Adapting the exact
-      ! same formulation as MAM now and tune later
-      ! As of NE-380: Oslo dust emissions are 2/3 of CAM emissions
-      ! gives better AOD close to dust sources
```
Is this comment really no longer relevant or accurate?

oyvindseland replied:
Line 177 and 178 is correct and can stay.
Line 179 and 180. > Suggest to change to: "The flux calculations from the Zender_2003 parameterisation  also include a second scaling factor of 1.15"
Line 181-182. Connected to a very specific model version so can be deleted I think.

gold2718 commented on src/oslo_aero_dust.F90:
```diff
@@ -113,7 +111,7 @@ subroutine oslo_aero_dust_readnl(nlfile)
             write(iulog, *) subname, ': dust_emis_fact = ', dust_emis_fact
          else
             !! XXgoldyXX: Remove this when Leung_2023 is implemented
```
This line should be removed.

gold2718 reviewed (changes requested):
I have no idea if this code works but your changes look good.
I do have a one change request and a question.

Commit:
address changes requested + fix missing import

This version of the leung changes do no compile sucessfully. Added back some of the comments i removed. Fix a typo in the log message.

Commit:
fix typo in comment

Ovewh commented:
I have compiled the code sucessfully, but not done any testing beyond that.

Ovewh commented:
I ran a few months to test and it seems to be working (setting `dust_emis_method = 'Leung_2023' ` in usr_nl_cam) I checked the dust optical depth for one month and got 0.12014, so there is definitely  some tuning needed.  Otherwise there is now larger dust optical depth over North Africa compared to Australia which is an improvement. I'm not that familiar with running NorESM3, so i just re-created [øyvinds case](https://github.com/NorESMhub/noresm3_dev_simulations/issues/124) . 
![output](https://github.com/user-attachments/assets/008405b7-14ec-4c7a-9032-4a7c9a775908)

oyvindseland commented:
The updated file is committed to the Oslo_Aero github repository, but it just lives there as a single commit.  From github 
"This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository."
Can a PR be made from this single file or should I make a branch based on exiting tag and then make a PR? @gold2718

gold2718 commented:
> The updated file is committed to the Oslo_Aero github repository, but it just lives there as a single commit. From github "This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository." Can a PR be made from this single file or should I make a branch based on exiting tag and then make a PR? @gold2718

I do not see this message or a problem. Where are you seeing this?

Ovewh commented:
> The updated file is committed to the Oslo_Aero github repository, but it just lives there as a single commit. From github "This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository." Can a PR be made from this single file or should I make a branch based on exiting tag and then make a PR? @gold2718

Yea it expected behavior, because the branch live in my fork and not in the Oslo_Aero repository.

oyvindseland commented:
No, it is not really a program. The important part is whether it can be used for the PR into the development branch or if I should create a new branch.

gold2718 reviewed (approved):
I think this is okay for testing with CAM.

oyvindseland commented:
I suggest to test the emissions in set-up starting from noresm3a03 + change oslo_aero and cam to their respective development  branches.
Is it possible that this may not work?

gold2718 commented:
> I suggest to test the emissions in set-up starting from noresm3a03 + change oslo_aero and cam to their respective development branches. Is it possible that this may not work?

I am not sure what you are proposing. Right now, the CAM development branch is the same as tag `noresm3_0_001_cam6_4_085`. To develop CAM with the new Oslo Aero tag, I suggest starting with these steps:

1. Check out a new branch starting at the top of NorESM/noresm_develop, e.g.,
    - `git fetch --tags <noresm_remote_name>`
    - `git checkout -b test_leung_dust <noresm_remote_name>/noresm_develop`
    - `./bin/git-fleximod update` (cross fingers)
2. Reset the Oslo Aero tag
    - Edit `.gitmodules` to change the Oslo Aero tag to `oslo_aero_3_0a003`
    - Try running `./bin/git-fleximod update` again, _or_
    - `cd src/chemistry/oslo_aero; git fetch --tags newremote.00; git checkout oslo_aero_3_0a003`

oyvindseland commented:
Have you already made a tag for oslo_aero including the new code for dust?

gold2718 commented:
> Have you already made a tag for oslo_aero including the new code for dust?

Yes, `oslo_aero_3_0a003`

oyvindseland commented:
I also had to update the CAM tag, since noresm_dev points to an older cam tag, but that was trivial following the same method as above

gold2718 commented:
> I also had to update the CAM tag, since noresm_dev points to an older cam tag, but that was trivial following the same method as above

Right, I forgot that we do not have a NorESM tag for the latest CAM yet.
